drivers/sensor/hdc20x0/hdc20x0.c

```diff
@@ -0,0 +1,246 @@
+#define hdc20x0_REG_TEMP_HIGH 0x01
+#define hdc20x0_REG_HUMIDITY_LOW 0x02
+#define hdc20x0_REG_HUMIDITY_HIGH 0x03
+#define hdc20x0_REG_INTERRUPT_DRDY 0x04
+#define hdc20x0_REG_TEMP_MAX 0x05
+#define hdc20x0_REG_HUMIDITY_MAX 0x06
+#define hdc20x0_REG_INTERRUPT_EN 0x07
+#define hdc20x0_REG_TEMP_OFFSET_ADJ 0x08
+#define hdc20x0_REG_HUMIDITY_OFFSET_ADJ 0x09
+#define hdc20x0_REG_TEMP_THR_L 0x0a
+#define hdc20x0_REG_TEMP_THR_H 0x0b
+#define hdc20x0_REG_RH_THR_L 0x0c
+#define hdc20x0_REG_RH_THR_H 0x0d
+#define hdc20x0_REG_RESET_DRDY_INT_CONF 0x0e
+#define hdc20x0_REG_MEASUREMENT_CONF 0x0f
+
+#define hdc20x0_MEAS_CONF GENMASK(2, 1)
+#define hdc20x0_MEAS_TRIG BIT(0)
+#define hdc20x0_HEATER_EN BIT(3)
+#define hdc20x0_AMM GENMASK(6, 4)
+
+struct hdc20x0_data {
+	const struct device *i2c_master;
+
+	/* Temperature Values */
+	int32_t temp_val1;
+	int32_t temp_val2;
+	/* Humidity Values*/
+	int32_t humidity_val1;
+	int32_t humidity_val2;
+};
+
+struct hdc20x0_cfg {
+	char *i2c_bus_name;
+	uint16_t i2c_addr;
+};
+
+static int
+hdc20x0_read_measurement_temperature(struct hdc20x0_data *data,
+				     const struct hdc20x0_cfg *config)
+{
+	uint8_t val;
+	float celsius;
+
+	if (i2c_reg_read_byte(data->i2c_master, config->i2c_addr,
+			      hdc20x0_REG_TEMP_LOW, &val)) {
+		LOG_ERR("Failed to get temperature LSB ");
+		return -EIO;
+	}
+	data->temp_val1 = val;
+	if (i2c_reg_read_byte(data->i2c_master, config->i2c_addr,
+			      hdc20x0_REG_TEMP_HIGH, &val)) {
+		LOG_ERR("Failed to get temperature MSB ");
+		return -EIO;
+	}
+	data->temp_val2 = val;
+	LOG_DBG("temperature,  val1:0x%x, val2:0x%x", data->temp_val1,
+		data->temp_val2);
+
+	/*https://www.ti.com/lit/ds/symlink/hdc20x0.pdf*/
+	/*7.6.2 Address 0x01 Temperature MSB*/
+	celsius = ((data->temp_val2 * 256 + data->temp_val1) / 65536.0) * 165 -
+		  40;
+	data->temp_val1 = (int)celsius;
+	data->temp_val2 = (celsius - data->temp_val1) * 1000000LL;
+	return 0;
+}
+
+static int hdc20x0_read_measurement_humidity(struct hdc20x0_data *data,
+					     const struct hdc20x0_cfg *config)
+{
+	uint8_t val;
+	float rh;
+
+	if (i2c_reg_read_byte(data->i2c_master, config->i2c_addr,
+			      hdc20x0_REG_HUMIDITY_LOW, &val)) {
+		LOG_ERR("Failed to get humidity LSB ");
+		return -EIO;
+	}
+	data->humidity_val1 = val;
+	if (i2c_reg_read_byte(data->i2c_master, config->i2c_addr,
+			      hdc20x0_REG_HUMIDITY_HIGH, &val)) {
+		LOG_ERR("Failed to get humidity MSB ");
+		return -EIO;
+	}
+	data->humidity_val2 = val;
+	LOG_DBG("humidity , val1:0x%x, val2:0x%x", data->humidity_val1,
+		data->humidity_val2);
+	/*https://www.ti.com/lit/ds/symlink/hdc20x0.pdf*/
+	/*7.6.4 Address 0x03 Humidity MSB*/
+	rh = ((data->humidity_val2 * 256 + data->humidity_val1) / 65536.0) *
+	     100;
+
+	data->humidity_val1 = (int)rh;
+	data->humidity_val2 = (rh - data->humidity_val1) * 1000000LL;
+	return 0;
+}
+/*measurement humidity and temperature*/
+static int hdc20x0_read_measurement_all(struct hdc20x0_data *data,
```

Comment:
hdc20xx_read_all(...)

Reply:
> Please check if HDC2010 support can be combined with [ti_hdc driver](https://github.com/zephyrproject-rtos/zephyr/tree/master/drivers/sensor/ti_hdc).

Sorry, they belong to different types of temperature and humidity sensors and cannot be compatible.

Reply:
Not really, register addressing is different but temperature/humidity calculation is still the same.